### PR TITLE
Add a Chip component

### LINKED ISSUE #24: Chip
Addressed by #21

### PR DESCRIPTION
**Pull request by blaketarter**
## Description

This PR adds a basic chip component used to show "chip" or "tag" like information. The component supports `onSelect` and `onClose` actions with keyboard support and a replaceable `closeElement` prop.

### 🖼 Demo

Screen Shot 2022-08-02 at 4 04 23 PM

https://user-images.githubusercontent.com/6101761/182473488-550c586e-90f7-4eab-8c55-5ea24dae883f.mov

## 🔎 Reviewer Checklist

> _Checked off by PR **Reviewers**_

- [ ] Merge destination is correct
- [ ] Code is correct as understood and conforms to quality standards
- [ ] Tests have been added where appropriate (unit, visual, end-to-end)
- [ ] Acceptance Criteria have been met

---

> ### Roles & Responsibilities
>
> #### 👨‍💻 Assignee
>
> - Initiator of this PR (be sure to set in GitHub UI)
> - Addresses feedback and change requests
> - Merges PR once approved (usually deletes branch unless `develop` or `release`)
>
> #### 👩‍💻 Reviewer
>
> - Invited to review PR by **Assignee** (via GitHub UI)
> - Is expected to complete a review and address followup

### CODE REVIEW

**Comment by blaketarter** [comment]
I'm torn on how much styling to include with something like this. I could add some more default styles to make it look better, or I could even remove all styles entirely. Anyone have any thoughts?

**Review by albatrocity — DISMISSED** [review]
Looks good to me. I'd say having an css module entry point like you have is enough to pave the way for "real" styling.

File: src/components/Chip/Chip.module.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.Chip {
```
Comment (roblafeve) [inline src/components/Chip/Chip.module.css]: Something I've noticed is that if we use the component name as the wrapper class, then css modules doubles up on it. I wonder if we should use a convention like `container` or something. That's what I did. 🤷 

Screen Shot 2022-08-02 at 5 21 11 PM

File: src/components/Chip/Chip.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { ReactNode } from "react"
+import styles from "./Chip.module.css"
+
+export interface ChipProps {
+  children?: ReactNode
```
Comment (roblafeve) [inline src/components/Chip/Chip.tsx]: Is `children` being optional a realistic scenario?

**Review by roblafeve — DISMISSED** [review]
Nice work!

**Review by belv2c — COMMENTED** [review]
Nice work! Since the chip is selectable and the closeElement is clickable, do you think it makes sense to add `cursor: pointer` to the css for these?

**Comment by thispastwinter** [comment]
> Nice work! Since the chip is selectable and the closeElement is clickable, do you think it makes sense to add `cursor: pointer` to the css for these?

+1

**Review by thispastwinter — COMMENTED** [review]
~~I personally think styling things a bit wouldn't hurt, if only to ensure that each respective element has a class tied to it. I'm trying to think of what would make this the most valuable if I were to pull it into a project. Knowing I could use things as is and quickly make some changes to various classes to get it to look good would be a huge boon.~~

I take that back, because you've basically given a great deal of freedom to enable someone to style it however they want by having a `closeElement` prop and a container that accepts a `ReactNode` child.

**Review by roblafeve — APPROVED** [review]
👌 :shipit:

**Review by belv2c — APPROVED** [review]
🚢

**Comment by roblafeve** [comment]
resolves #24

**Pull request merged**